packages/evm/contracts/budgets/TransparentBudget.sol

```diff
@@ -0,0 +1,339 @@
+import {ATransparentBudget} from "contracts/budgets/ATransparentBudget.sol";
+
+/*
+    TODO
+    1. test deposit and boost creation logic
+    2. implement clawback logic and tracking on deposits
+    3. implement clawback auth
+    4. implement permit2 support
+*/
+
+/// @title Simple ABudget
+/// @notice A minimal budget implementation that simply holds and distributes tokens (ERC20-like and native)
+/// @dev This type of budget supports ETH, ERC20, and ERC1155 assets only
+contract TransparentBudget is ATransparentBudget, ReentrancyGuard {
+    using SafeTransferLib for address;
+    using DynamicArrayLib for *;
+    using LibTransient for *;
+
+    /// @dev The total amount of each fungible asset distributed from the budget
+    mapping(address => uint256) private _distributedFungible;
+
+    /// @dev The total amount of each ERC1155 asset and token ID distributed from the budget
+    mapping(address => mapping(uint256 => uint256)) private _distributedERC1155;
+
+    /// @inheritdoc ABudget
+    /// @notice This is unused. Call `createBoost` with a deposit payload
+    function allocate(bytes calldata) external payable virtual override returns (bool) {
+        revert BoostError.NotImplemented();
+    }
+
+    function createBoost(bytes[] calldata _allocations, BoostCore core, bytes calldata _boostPayload)
+        external
+        payable
+        nonReentrant
+    {
+        DynamicArrayLib.DynamicArray memory allocationKeys;
+        allocationKeys.reserve(_allocations.length);
+
+        for (uint256 i = 0; i < _allocations.length; i++) {
+            bytes32 key = _allocate(_allocations[i]);
+            allocationKeys.set(i, key);
+        }
+
+        core.createBoost(_boostPayload);
+
+        bytes32[] memory keys = allocationKeys.asBytes32Array();
+        for (uint256 i = 0; i < keys.length; i++) {
+            LibTransient.TUint256 storage p = LibTransient.tUint256(keys[i]);
+            if (p.get() != 0) revert BoostError.Unauthorized();
+        }
+    }
+
+    /// @notice Allocates assets to be distributed in the boost
+    /// @param data_ The packed data for the {Transfer} request
+    /// @return key The key of the amount allocated
+    /// @dev The caller must have already approved the contract to transfer the asset
+    /// @dev If the asset transfer fails, the allocation will revert
+    function _allocate(bytes calldata data_) internal virtual returns (bytes32 key) {
+        Transfer memory request = abi.decode(data_, (Transfer));
+        if (request.assetType == AssetType.ETH) {
+            FungiblePayload memory payload = abi.decode(request.data, (FungiblePayload));
+
+            // Ensure the value received is equal to the `payload.amount`
+            if (msg.value != payload.amount) {
+                revert InvalidAllocation(request.asset, payload.amount);
+            }
+            (LibTransient.TUint256 storage p, bytes32 tKey) = getFungibleAmountAndKey(address(0));
+            p.inc(payload.amount);
```

Comment:
Okay so here's where we latch in - this is triggered in the top half of the `createBoost` call and serves as the "latch on"

Just musing no actionables.